src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/ProxyAuthenticationHandler.cs

```diff
@@ -103,6 +102,12 @@ public class ProxyAuthenticationHandler : DelegatingHandler
         // Returns true if the cause of the exception is proxy authentication failure
         private static bool ProxyAuthenticationRequired(Exception ex)
         {
+            // HACK!!! : This is a hack to workaround Xamarin Bug 19594
+            if (RuntimeEnvironmentHelper.IsMono && (ex is WebException))
+            {
+                return IsMonoProxyAuthenticationRequiredError((WebException) ex);
```

Comment:
nit: this should be cast one time, code analysis use to complain about this but I don't think that check is done in our current build